pkg/git/repository.go

```diff
@@ -192,9 +192,9 @@ func (o *CloneOptions) Path() string {
 	return o.path
 }
 
-func (r *repo) Persist(ctx context.Context, opts *PushOptions) error {
+func (r *repo) Persist(ctx context.Context, opts *PushOptions) (string, error) {
```

Comment:
I think this might be the perfect time to split this to `Commit()` which returns a revision and `Persist()`, which would push all commits. This would later allow us to advance towards #114

Reply:
If I currently split it into `Commit` and `Persist` (which is actually `Push`), I will have to refactor all the call sites to call them both (since everywhere we currently always do commit + push).
I think we should have a different PR and user story that separates the two, and only pushes on a specific flag. But it's not related to this PR.